### PR TITLE
feat: Sync with Cloud

### PR DESCRIPTION
**Pull request by devcatalin**
This PR...

## Changes

- added `@monokle/synchronizer` for authentication and policy fetching from Cloud
- updated validation to use cloud policy when found
- added Cloud icon in the page header
- set validation settings as readonly + policy banner to notify users about it (when using cloud policy)

## Checklist

- [x] tested locally
- [x] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: electron/app/services/cloud/login.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  }
+  const loginResponse = await authenticator.login('device code');
+  if (!loginResponse.handle) {
+    throw new Error('Something went wrong with the login response');
+  }
+  shell.openExternal(loginResponse.handle.verification_uri_complete);
+  const user = await loginResponse.onDone;
+  if (!user) {
+    throw new Error('Login to Cloud has failed');
+  }
```
Comment (olensmar) [inline electron/app/services/cloud/login.ts:21]: can we get any more context/details for any of the above error conditions and write those to the logs!?

File: src/components/organisms/PageHeader/CloudSync/CloudSync.tsx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+            </Title>
+            <Value>{cloudUser.email}</Value>
+          </Item>
+          <Item>
+            <Title>{projectInfo?.name ? <GreenCircle /> : <RedCircle />} Cloud project</Title>
+            <Value>
+              {projectInfo?.name ? (
+                <>
+                  <Link onClick={() => openUrlInExternalBrowser(projectInfo.link)}>{projectInfo.name}</Link>
+                  <Info description="The Monokle Cloud project manages this repository." />
```
Comment (olensmar) [inline src/components/organisms/PageHeader/CloudSync/CloudSync.tsx:58]: "A Monokle Cloud Project manages this repository"

File: src/components/organisms/PageHeader/CloudSync/CloudSync.tsx [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+            <Value>
+              {cloudPolicy ? (
+                <>
+                  <Link onClick={() => openUrlInExternalBrowser(policyInfo?.link)}>View project policy</Link>
+                  <Info description="A Policy is configured for this repository in Monokle Cloud. Your local Policy is in-sync and cannot be changed." />
+                </>
+              ) : (
+                <>
+                  <span>Not found</span>
+                  <Info description="Couldn't find any Policy configured for this repository." />
```
Comment (olensmar) [inline src/components/organisms/PageHeader/CloudSync/CloudSync.tsx:79]: Can we add a [Configure Policy] link that goes to the policy builder?

File: src/components/organisms/SettingsPane/CloudConnect/CloudConnect.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        <S.Row>
+          <Col span={12} style={{padding: '8px'}}>
+            {cloudUser ? (
+              <>
+                <S.Heading>Connected to Monokle Cloud</S.Heading>
+                <S.Paragraph>
+                  E-mail: <b>{cloudUser.email}</b>
+                </S.Paragraph>
+                <S.Paragraph>
+                  Thank you for syncing Monokle Desktop with Monokle Cloud. <br />
```
Comment (olensmar) [inline src/components/organisms/SettingsPane/CloudConnect/CloudConnect.tsx:35]: "You have successfully connected to your Monokle Cloud Account."

**Review by olensmar — COMMENTED** [review]
All good - just a couple of wording updates..

File: src/components/organisms/PageHeader/CloudSync/CloudSync.tsx (hunk 3 above)
Comment (devcatalin) [reply src/components/organisms/PageHeader/CloudSync/CloudSync.tsx:79]: I can't add this easily right now, we can try it in a follow up after I talk to someone from Cloud

**Review by topliceanurazvan — APPROVED** [review]

**Pull request merged**